File: GraphMigrator/GraphMigrator.Domain/Models/EnhancedQueryComparisonResult.cs
namespace GraphMigrator.Domain.Models;

public class EnhancedQueryComparisonResult : QueryComparisonResult
{
    public SqlPerformanceMetrics SqlPerformanceMetrics { get; set; }
    public Neo4jPerformanceMetrics Neo4jPerformanceMetrics { get; set; }
}

File: GraphMigrator.UI/GraphMigratorClient/src/app/models/enhanced-query-comparison-result.ts
import { Neo4jPerformanceMetrics } from './neo4j-performance-metrics';
import { QueryComparisonResult } from './query-comparison-result';
import { SqlPerformanceMetrics } from './sql-performance-metrics';

export interface EnhancedQueryComparisonResult extends QueryComparisonResult {
  sqlPerformanceMetrics: SqlPerformanceMetrics;
  neo4jPerformanceMetrics: Neo4jPerformanceMetrics;
}
